runtime/drivers/mysql/olap.go

```diff
@@ -0,0 +1,338 @@
+func (c *connection) LoadPhysicalSize(ctx context.Context, tables []*drivers.OlapTable) error {
+	return nil
+}
+
+// Lookup implements drivers.OLAPInformationSchema.
+func (c *connection) Lookup(ctx context.Context, db, schema, name string) (*drivers.OlapTable, error) {
+	meta, err := c.GetTable(ctx, db, schema, name)
+	if err != nil {
+		return nil, err
+	}
+
+	rtSchema := &runtimev1.StructType{}
+	for name, typ := range meta.Schema {
+		t, err := databaseTypeToPB(typ, true)
+		if err != nil {
+			return nil, err
+		}
+		rtSchema.Fields = append(rtSchema.Fields, &runtimev1.StructType_Field{
+			Name: name,
+			Type: t,
+		})
+	}
+	return &drivers.OlapTable{
+		Database:          db,
+		DatabaseSchema:    schema,
+		Name:              name,
+		View:              meta.View,
+		Schema:            rtSchema,
+		UnsupportedCols:   nil,
+		PhysicalSizeBytes: 0,
+	}, nil
+}
+
+func rowsToSchema(r *sqlx.Rows) (*runtimev1.StructType, error) {
+	if r == nil {
+		return nil, nil
+	}
+
+	cts, err := r.ColumnTypes()
+	if err != nil {
+		return nil, err
+	}
+
+	fields := make([]*runtimev1.StructType_Field, len(cts))
+	for i, ct := range cts {
+		nullable, ok := ct.Nullable()
+		if !ok {
+			nullable = true
+		}
+
+		t, err := databaseTypeToPB(ct.DatabaseTypeName(), nullable)
+		if err != nil {
+			return nil, err
+		}
+
+		fields[i] = &runtimev1.StructType_Field{
+			Name: ct.Name(),
+			Type: t,
+		}
+	}
+
+	return &runtimev1.StructType{Fields: fields}, nil
+}
+
+func databaseTypeToPB(dbt string, nullable bool) (*runtimev1.Type, error) {
+	t := &runtimev1.Type{Nullable: nullable}
+	switch strings.ToUpper(dbt) {
+	case "DECIMAL":
+		t.Code = runtimev1.Type_CODE_STRING
+	case "NUMERIC":
+		t.Code = runtimev1.Type_CODE_DECIMAL
+	case "BIT":
+		t.Code = runtimev1.Type_CODE_STRING
+	case "TINYINT":
+		t.Code = runtimev1.Type_CODE_INT8
+	case "SMALLINT":
+		t.Code = runtimev1.Type_CODE_INT16
+	case "MEDIUMINT":
+		t.Code = runtimev1.Type_CODE_INT32
+	case "INT", "INTEGER":
+		t.Code = runtimev1.Type_CODE_INT32
+	case "BIGINT":
+		t.Code = runtimev1.Type_CODE_INT64
+	case "UNSIGNED BIGINT":
+		t.Code = runtimev1.Type_CODE_INT64
+	case "BOOLEAN", "BOOL":
+		t.Code = runtimev1.Type_CODE_BOOL
+	case "FLOAT", "DOUBLE", "REAL":
+		t.Code = runtimev1.Type_CODE_FLOAT64
+	case "CHAR", "VARCHAR", "BINARY", "VARBINARY", "TINYBLOB", "BLOB", "MEDIUMBLOB", "LONGBLOB", "TINYTEXT", "TEXT", "MEDIUMTEXT", "LONGTEXT":
+		t.Code = runtimev1.Type_CODE_STRING
+	case "ENUM", "SET":
+		t.Code = runtimev1.Type_CODE_STRING
+	case "DATE":
+		t.Code = runtimev1.Type_CODE_DATE
+	case "TIME":
+		t.Code = runtimev1.Type_CODE_TIME
+	case "DATETIME", "TIMESTAMP":
+		t.Code = runtimev1.Type_CODE_TIMESTAMP
+	case "YEAR":
+		t.Code = runtimev1.Type_CODE_INT16
+	case "JSON":
+		t.Code = runtimev1.Type_CODE_JSON
+	case "GEOMETRY":
+		t.Code = runtimev1.Type_CODE_STRING
+	case "NULL":
+		t.Code = runtimev1.Type_CODE_UNSPECIFIED
+	default:
+		return nil, fmt.Errorf("unhandled MySQL type: %s", dbt)
```

Comment:
Would the unspecified type be better? Or do you feel good that we covered all data types? I'm not that familiar with how stable MySQL's types are.